### PR TITLE
feat(activity-logging): track UI actions on persons

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

The UI lets a user make edits to persons. We don't know who changed what...

## Changes

In the UI you can

* add properties to a person
* merge several people
* start a task to split people
* delete a person

This adds log_activity calls to each of these activities.

## How did you test this code?

adding API level unit tests. and making changes in the UI and observing the records in the database.

### CODE REVIEW

File: posthog/api/person.py [hunk 1]
```diff
@@ -508,6 +572,52 @@ def cohorts(self, request: request.Request) -> response.Response:
...
+    def activity(self, request: request.Request, **kwargs):
+        item_id = kwargs["pk"]
+        if not self.get_queryset().filter(id=item_id, team_id=self.team_id).exists():
+            return Response("", status=status.HTTP_404_NOT_FOUND)
+
+        activity = load_activity(scope="Person", team_id=self.team_id, item_id=item_id,)
+        return Response(
+            {"results": ActivityLogSerializer(activity, many=True,).data, "next": None, "previous": None,},
+            status=status.HTTP_200_OK,
+        )
```
Comment (Twixes) [inline posthog/api/person.py:593]: These two actions seem quite boilerplate-y, having just reviewed the PR around feature flags. Could they go into some mixin? Looks like `scope` is the only customizable part.

File: posthog/models/activity_logging/activity_log.py [hunk 2]
```diff
@@ -33,6 +42,12 @@ def default(self, obj):
             return obj.__dict__
         if isinstance(obj, Change):
             return obj.__dict__
+        if isinstance(obj, Merge):
+            return obj.__dict__
```
Comment (Twixes) [inline posthog/models/activity_logging/activity_log.py:46]: These three `isinstance`s can be consolidated into one: `isinstance(obj, (Detail, Change, Merge))`

**Review by Twixes — COMMENTED** [review]
Looks pretty good

File: posthog/models/activity_logging/activity_log.py (hunk 2 above)
Comment (pauldambra) [reply posthog/models/activity_logging/activity_log.py:46]: Oh, that's cool!

File: posthog/api/person.py (hunk 1 above)
Comment (pauldambra) [reply posthog/api/person.py:593]: Hmmm 🤔 

There's the existence check too... although possible that could be removed now there's no history hog (RIP)

They are boilerplate, but also, only two examples currently and more coming. So I'd suggest we say "a little duplication is better than the wrong abstraction" and push this decision into the Insights implementation of this

Comment (Twixes) [reply posthog/api/person.py:593]: Sure, _Rule of Three_!

**Review by Twixes — APPROVED** [review]

**Pull request merged**